predicators/datasets/vlm_input_data.py

```diff
@@ -1,17 +1,245 @@
 """Functions to create offline demonstration data by leveraging VLMs."""
 
+import ast
+import glob
 import logging
 import os
 import re
-from typing import Dict, List, Sequence, Set, Tuple
+from pathlib import Path
+from typing import Dict, List, Optional, Sequence, Set, Tuple
 
 import numpy as np
+import PIL.Image
 
 from predicators.envs import BaseEnv
+from predicators.envs.vlm_envs import VLMPredicateEnv
+from predicators.pretrained_model_interface import GoogleGeminiVLM, \
+    VisionLanguageModel
 from predicators.settings import CFG
 from predicators.structs import Action, Dataset, GroundAtom, \
-    LowLevelTrajectory, Object, ParameterizedOption, Predicate, State, Task, \
-    _Option
+    ImageOptionTrajectory, LowLevelTrajectory, Object, ParameterizedOption, \
+    Predicate, State, Task, _Option
+
+
+def _generate_prompt_for_atom_proposals(
+        traj: ImageOptionTrajectory, trajectory_subsample_freq: int
+) -> List[Tuple[str, List[PIL.Image.Image]]]:
```

Comment:
what's the reason for using PIL instead of our regular `structs.Image`?

Reply:
Gemini consumes `PIL.Image` directly! So just made life a bit easier